### ISSUE TITLE
bug: Animation jank and overscroll issues when using IonRipple within an IonTabs iOS application

### ISSUE THREAD

**Issue by uncvrd** [ionic-team/ionic-framework#21866]
# Bug Report

**Ionic version:**

[x] **5.x**

**Current behavior:**

I've discovered some odd animation jank and overscroll problems when using List items that contain an `IonRipple` in a Tabs application. Sometimes, when clicking on an item, it causes the list to shift down and you can no longer access the last item in the list (it gets shifted beneath the tab bar) Here's how to recreate (timestamps of my recreation below instructions):

Link: https://streamable.com/9k8qi9

- When scrolling to the bottom of a list of items, scroll to the last item
- Stop where half of the last item is showing above the tab bar. 
- Now click on that item (page routes forward) and then swipe back. 
- If you try tapping on a new item in the list you'll notice that the list gets shifted downwards and if you try swiping all the down to get to the bottom of the list, you can no longer do so (you can see the the last item due to overscrolling/rubberband scrolling on iOS). 
- If you remove the `IonRipple` element from the last item, this no longer happens which leads me to believe something from the ripple animation is causing an issue with some sort of offset height in the `IonContent`

I've recreated this on the iOS simulator and on a physical iOS device, here are detailed timestamps of my video above:

**0:00-0:04** - I scroll to the bottom of the list - lining up the last item to be halfway above the tab bar

**0:04-0:06** - I route forward and back to the list

**0:06-0:09** - I scroll to the bottom of the list (Kelly Richardson) to show that I can hit the bottom still, then line Kelly Richardson halfway between the tab bar (not sure if that's really relevant)

**0:10** - I tap on another element in the list and the screen snaps down a bit (the animation jank I was talking about)

**0:10-0:16** - I now show that I can no longer scroll to the bottom of the list
 
Here's the repo that I used (which is just the base Ionic List starter app with Tabs added on the App.tsx): https://github.com/uncvrd/ionic-animation-with-tabs

```
Ionic:

   Ionic CLI       : 6.7.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 5.3.1

Capacitor:

   Capacitor CLI   : 2.4.0
   @capacitor/core : 2.4.0

Utility:

   cordova-res (update available: 0.15.1) : 0.14.0
   native-run                             : 1.0.0

System:

   NodeJS : v12.16.2 (/usr/local/bin/node)
   npm    : 6.14.4
   OS     : macOS Catalina
```

---

Edit: Also if you look closely at the video link...juuuuust as the page transition animation is finishing up (when I route forward a page) you can see that the list shifts up a tiny bit (look at the left side of the screen as the sliding animation finishes). This list shifting is noticeable when I swipe back because I have to swipe down to get where I was in the list afterwards

Edit 2: Here's an additional video link of the same issue on my personal app. Switching tabs after the issue shows up fixes the problem until the ripple effect is triggered again while scrolling: https://streamable.com/srf6nr

**Comment by liamdebeasi** [comment]
Thanks for the issue. I am unable to reproduce the issue using the code provided. Can you double check your repo to make sure everything is there that I need?

**Comment by uncvrd** [comment]
Hey @liamdebeasi yes, I've confirmed all the code is there! I recreated it again this morning after a clean clone of the repository I've linked using both my iOS simulator and my physical device. Here's a video of it this morning where I don't do anything special: https://streamable.com/ycksjq

I just scrolled to the very bottom, noted that I see Kelly Richardson, clicked on her email, swiped back, and now the list view doesn't show Kelly at the bottom of the list (it got shifted down) and when I try to swipe further, I can only rubberband scroll and see her beneath Moe Chamont

**Comment by liamdebeasi** [comment]
Thanks, I was able to reproduce the issue. I did not notice that the IonRippleEffect component was commented out the first time around. I am going to close this as this is not a bug in Ionic Framework. This issue appears to be related to a WebKit bug that I filed a few months ago: https://bugs.webkit.org/show_bug.cgi?id=209968.

In this case, the ripple effect is causing the scroll position to get reset when rubber banding. It looks like it also has something to do with some of the items getting stuck off screen.

As a temporary workaround, I recommend not using the ripple effect on `ion-item` until this issue is resolved by the WebKit team. Thanks!
